comixed-tasks/src/main/java/org/comixedproject/task/runner/TaskManager.java

```diff
@@ -43,26 +44,45 @@
   @Autowired private ThreadPoolTaskExecutor taskExecutor;
   @Autowired private TaskService taskService;
 
-  public void runTask(final WorkerTask task) {
+  /**
+   * Enqueues a task, then deletes it upon successful completion.
+   *
+   * @param workerTask the worker task
+   * @param task the persisted task
+   */
+  public void runTask(final WorkerTask workerTask, final Task task) {
     this.taskExecutor.execute(
         () -> {
-          final String description = task.getDescription();
+          final String description = workerTask.getDescription();
           log.debug("Preparing to run task: {}", description);
           final Date started = new Date();
           try {
-            task.startTask();
-            if (!(task instanceof MonitorTaskQueueWorkerTask))
+            workerTask.startTask();
+            if (!(workerTask instanceof MonitorTaskQueueWorkerTask))
               this.updateAuditLog(true, started, description, null);
           } catch (WorkerTaskException error) {
             log.error("Error executing task: {}" + description, description, error);
-            if (!(task instanceof MonitorTaskQueueWorkerTask))
+            if (!(workerTask instanceof MonitorTaskQueueWorkerTask))
               this.updateAuditLog(false, started, description, error);
           } finally {
-            task.afterExecution();
+            workerTask.afterExecution();
+            if (task != null) {
+              log.debug("Deleting persisted task");
+              this.taskService.delete(task);
+            }
           }
         });
   }
 
+  /**
+   * Enqueue and execute the given task.
+   *
+   * @param task the task
+   */
+  public void runTask(final WorkerTask task) {
```

Comment:
D'accord. Je l'ai renommé.

Reply:
ho sorry ... I wrote in French !!!! :D 
For non French speaking people :
workerTask would be better than task to avoid confusion (between the WorkerTask class and the Task class).